From 797032c78aeb3b7e263eabdc1a3376ee494e699e Mon Sep 17 00:00:00 2001
Message-Id: <797032c78aeb3b7e263eabdc1a3376ee494e699e.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Damien Lespiau <damien.lespiau@intel.com>
Date: Sat, 30 Aug 2014 16:51:03 +0100
Subject: [PATCH 279/312] FROM_UPSTREAM [VPG]: drm/i915: Don't restrict
 i915_wa_registers to BDW

We have CHV code that already makes the test obsolete. Besides, when
num_wa_regs is 0 (platforms not gathering that W/A data), we expose
something sensible already.

Issue: VIZ-4092
Issue: GMIN-3475
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: fd34f90c6071bbc124572dae83b67044b1c663cc
Change-Id: I1254f97314d74f2b4f5988776ec2840b4ed36c40
Signed-off-by: Damien Lespiau <damien.lespiau@intel.com>
Reviewed-by: Arun Siluvery <arun.siluvery@linux.intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 01a085a..07c7c23 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -3088,11 +3088,6 @@ static int i915_wa_registers(struct seq_file *m, void *unused)
 	struct drm_device *dev = node->minor->dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
-	if (!IS_BROADWELL(dev)) {
-		DRM_DEBUG_DRIVER("Workaround table not available !!\n");
-		return -EINVAL;
-	}
-
 	ret = mutex_lock_interruptible(&dev->struct_mutex);
 	if (ret)
 		return ret;
-- 
1.7.9.5

